src/pdf/pdf.js

```diff
@@ -13,6 +12,12 @@ async function urlToPdfString(url, options) {
 }
 
 async function savePdfFile(filepath, url, options) {
+  const pdfDir = path.dirname(filepath);
+  try {
+    await fs.promises.access(pdfDir, fs.constants.F_OK);
+  } catch {
+    await fs.promises.mkdir(pdfDir, { recursive: true });
+  }
```

Comment:
```suggestion
  const pdfDir = path.dirname(filepath);
  await fs.promises.mkdir(pdfDir, { recursive: true });
```

`mkdir` with `recursive: true` won't fail if the directory exists so there is no need to check the existence first (see [doc](https://nodejs.org/dist/latest-v14.x/docs/api/fs.html#fs_fs_mkdir_path_options_callback)). It's not a very good idea anyway because there is no guarantee that the directory won't be deleted between the check (`access`) and the action that depends on it (`writeFile`).